lm_eval/api/model.py

```diff
@@ -303,11 +304,19 @@ def _encode_pair(self, context, continuation):
             continuation = context[-n_spaces:] + continuation
             context = context[:-n_spaces]
 
-        whole_enc = self.tok_encode(context + continuation)
-        context_enc = self.tok_encode(context)
+        if self.AUTO_MODEL_CLASS == transformers.AutoModelForCausalLM:
```

Comment:
If I am not mistaken `self.AUTO_MODEL_CLASS` is not defined inside `TemplateLM` or `LM`, so this check will probably cause error if `self.AUTO_MODEL_CLASS` will not be defined in descendant class.

Reply:
Sure! For openai-completions models that inherit from TemplateLM attr `AUTO_MODEL_CLASS` is not defined which caused an error till revert of the previous seq2seq patch. This one should either pass `AUTO_MODEL_CLASS` directly into TemplateLM or not use this attr at all.

Reply:
I think the best approach here would be to leave a comment that this should be overridden for Seq2Seq models and to have HFLM use this version of `_encode_pair()`.